pkg/storage/backend-storage/backend-storage.go

```diff
@@ -41,12 +44,95 @@ import (
 )
 
 const (
-	PVCPrefix = "persistent-state-for-"
+	PVCPrefix = "persistent-state-for"
 	PVCSize   = "10Mi"
 )
 
-func PVCForVMI(vmi *corev1.VirtualMachineInstance) string {
-	return PVCPrefix + vmi.Name
+func BasePVC(vmi *corev1.VirtualMachineInstance) string {
+	return PVCPrefix + "-" + vmi.Name
+}
+
+func PVCForVMI(pvcStore cache.Store, vmi *corev1.VirtualMachineInstance, client ...kubecli.KubevirtClient) *v1.PersistentVolumeClaim {
+	var legacyPVC *v1.PersistentVolumeClaim
+
+	objs := pvcStore.List()
+	for _, obj := range objs {
+		pvc := obj.(*v1.PersistentVolumeClaim)
+		if pvc.Namespace != vmi.Namespace {
+			continue
+		}
+		vmName, found := pvc.Labels[PVCPrefix]
+		if found && vmName == vmi.Name {
+			return pvc
+		}
+		if pvc.Name == BasePVC(vmi) {
+			legacyPVC = pvc
+		}
+	}
+
+	if legacyPVC != nil {
+		if len(client) == 1 {
+			labelLegacyPVC(client[0], legacyPVC, vmi.Name)
+		}
+		return legacyPVC
+	}
+
+	return nil
+}
+
+func pvcForMigrationTargetFromStore(pvcStore cache.Store, migration *corev1.VirtualMachineInstanceMigration) *v1.PersistentVolumeClaim {
+	objs := pvcStore.List()
+	for _, obj := range objs {
+		pvc := obj.(*v1.PersistentVolumeClaim)
+		if pvc.Namespace != migration.Namespace {
+			continue
+		}
+		migrationName, found := pvc.Labels[corev1.MigrationNameLabel]
+		if found && migrationName == migration.Name {
+			return pvc
+		}
+	}
+
+	return nil
+
+}
+
+func PVCForMigrationTarget(pvcStore cache.Store, migration *corev1.VirtualMachineInstanceMigration) *v1.PersistentVolumeClaim {
+	if migration.Status.MigrationState != nil && migration.Status.MigrationState.TargetPersistentStatePVCName != "" {
+		obj, exists, err := pvcStore.GetByKey(migration.Namespace + "/" + migration.Status.MigrationState.TargetPersistentStatePVCName)
+		if err != nil || !exists {
+			return nil
+		}
+		return obj.(*v1.PersistentVolumeClaim)
+	}
+
+	return pvcForMigrationTargetFromStore(pvcStore, migration)
+}
+
+func labelLegacyPVC(client kubecli.KubevirtClient, pvc *v1.PersistentVolumeClaim, name string) {
+	labelPatch := patch.New()
+	if len(pvc.Labels) == 0 {
+		labelPatch.AddOption(patch.WithAdd("/metadata/labels", map[string]string{PVCPrefix: name}))
+	} else {
+		labelPatch.AddOption(patch.WithReplace("/metadata/labels/"+PVCPrefix, name))
+	}
+	labelPatchPayload, err := labelPatch.GeneratePayload()
+	if err == nil {
+		_, err = client.CoreV1().PersistentVolumeClaims(pvc.Namespace).Patch(context.Background(), pvc.Name, types.JSONPatchType, labelPatchPayload, metav1.PatchOptions{})
+		if err != nil {
+			log.Log.Reason(err).Warningf("failed to label legacy PVC %s/%s", pvc.Namespace, pvc.Name)
+		}
+	}
+}
+
+func CurrentPVCName(vmi *corev1.VirtualMachineInstance) string {
```

Comment:
Hey @jean-edouard, I'm working on some helper functions to enumerate all VM/VMI related volumes (including backend): https://github.com/kubevirt/kubevirt/pull/13096. This behavior change would make it impossible to access the backend storage volume if we don't have access to the VMI status. Could we find some way to include this PVC name in the VM status too? Thanks!

Reply:
I'm not sure how we would do that, and if it would make sense...
However, `PVCForVMI()` can be made generic to be called with either a VM or a VMI, since it just uses the name and namespace. Would that solve your issue?
Thanks!

Reply:
I don’t think it would work because that function still requires a client and the PVC cache, which might not be available everywhere. The helpers are designed to be used throughout all code, especially in snapshot, export, and clone-related operations, where we need access to backend volumes. Having a PVC with a randomly generated name that's only accessible if you have access to the VMI status will add complexity and may make it impossible in some places to access the volume. I'm trying to think of some way to address this as I don’t intend to block this PR.